DateMathExpressionResolverTests fails on master

Three tests have started failing over the weekend, I believe it has to do with the end of the year approaching. Not sure if the failures may be related to datetime changes. They happen only in master:

```
org.elasticsearch.cluster.metadata.DateMathExpressionResolverTests testExpression_CustomFormat
org.elasticsearch.cluster.metadata.DateMathExpressionResolverTests testExpression_CustomTimeZoneInIndexName
org.elasticsearch.cluster.metadata.DateMathExpressionResolverTests testExpression_EscapeDateFormat
```

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+periodic/447/console


They easily reproduce:

```
06:36:03   2> REPRODUCE WITH: ./gradlew :server:unitTest -Dtests.seed=57AEC52CB025A748 -Dtests.class=org.elasticsearch.cluster.metadata.DateMathExpressionResolverTests -Dtests.method="testExpression_EscapeDateFormat" -Dtests.security.manager=true -Dtests.locale=da-DK -Dtests.timezone=Indian/Mayotte -Dcompiler.java=11 -Druntime.java=8
06:36:03   2> REPRODUCE WITH: ./gradlew :server:unitTest -Dtests.seed=57AEC52CB025A748 -Dtests.class=org.elasticsearch.cluster.metadata.DateMathExpressionResolverTests -Dtests.method="testExpression_CustomFormat" -Dtests.security.manager=true -Dtests.locale=da-DK -Dtests.timezone=Indian/Mayotte -Dcompiler.java=11 -Druntime.java=8
06:36:03   2> REPRODUCE WITH: ./gradlew :server:unitTest -Dtests.seed=57AEC52CB025A748 -Dtests.class=org.elasticsearch.cluster.metadata.DateMathExpressionResolverTests -Dtests.method="testExpression_CustomTimeZoneInIndexName" -Dtests.security.manager=true -Dtests.locale=da-DK -Dtests.timezone=Indian/Mayotte -Dcompiler.java=11 -Druntime.java=8
```

```
06:36:03 FAILURE 0.02s J5 | DateMathExpressionResolverTests.testExpression_EscapeDateFormat <<< FAILURES!
06:36:03    > Throwable #1: java.lang.AssertionError: 
06:36:03    > Expected: ".marvel-{year}2018"
06:36:03    >      but: was ".marvel-{year}2019"
06:36:03    > 	at __randomizedtesting.SeedInfo.seed([57AEC52CB025A748:C795FA0EBA4BCE4B]:0)
06:36:03    > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
06:36:03    > 	at org.elasticsearch.cluster.metadata.DateMathExpressionResolverTests.testExpression_EscapeDateFormat(DateMathExpressionResolverTests.java:111)
06:36:03 FAILURE 0.00s J5 | DateMathExpressionResolverTests.testExpression_CustomFormat <<< FAILURES!
06:36:03    > Throwable #1: java.lang.AssertionError: 
06:36:03    > Expected: ".marvel-2018.12.30"
06:36:03    >      but: was ".marvel-2019.12.30"
06:36:03    > 	at __randomizedtesting.SeedInfo.seed([57AEC52CB025A748:6DA2B3B50CD96D0C]:0)
06:36:03    > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
06:36:03    > 	at org.elasticsearch.cluster.metadata.DateMathExpressionResolverTests.testExpression_CustomFormat(DateMathExpressionResolverTests.java:97)
06:36:03 FAILURE 0.01s J5 | DateMathExpressionResolverTests.testExpression_CustomTimeZoneInIndexName <<< FAILURES!
06:36:03    > Throwable #1: java.lang.AssertionError: 
06:36:03    > Expected: ".marvel-2018.12.30"
06:36:03    >      but: was ".marvel-2019.12.30"
06:36:03    > 	at __randomizedtesting.SeedInfo.seed([57AEC52CB025A748:C7B07BB86A595B07]:0)
06:36:03    > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
06:36:03    > 	at org.elasticsearch.cluster.metadata.DateMathExpressionResolverTests.testExpression_CustomTimeZoneInIndexName(DateMathExpressionResolverTests.java:153)
```

--- a/server/src/test/java/org/elasticsearch/action/admin/indices/rollover/RolloverIT.java
+++ b/server/src/test/java/org/elasticsearch/action/admin/indices/rollover/RolloverIT.java
@@ -272,11 +272,10 @@ public class RolloverIT extends ESIntegTestCase {
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/37037")
     public void testRolloverWithDateMath() {
         ZonedDateTime now = ZonedDateTime.now(ZoneOffset.UTC);
         assumeTrue("only works on the same day", now.plusMinutes(5).getDayOfYear() == now.getDayOfYear());
-        String index = "test-" + DateFormatter.forPattern("YYYY.MM.dd").format(now) + "-1";
+        String index = "test-" + DateFormatter.forPattern("yyyy.MM.dd").format(now) + "-1";
         String dateMathExp = "<test-{now/d}-1>";
         assertAcked(prepareCreate(dateMathExp).addAlias(new Alias("test_alias")).get());
         ensureGreen(index);
@@ -284,34 +283,34 @@ public class RolloverIT extends ESIntegTestCase {
         client().admin().indices().prepareClose(index).get();
         client().admin().indices().prepareUpdateSettings(index).setSettings(Settings.builder()
             .put(IndexMetaData.SETTING_INDEX_PROVIDED_NAME,
-            "<test-{now/M{YYYY.MM}}-1>")).get();
+            "<test-{now/M{yyyy.MM}}-1>")).get();
 
         client().admin().indices().prepareOpen(index).get();
         ensureGreen(index);
         RolloverResponse response = client().admin().indices().prepareRolloverIndex("test_alias").get();
         assertThat(response.getOldIndex(), equalTo(index));
-        assertThat(response.getNewIndex(), equalTo("test-" + DateFormatter.forPattern("YYYY.MM").format(now) + "-000002"));
+        assertThat(response.getNewIndex(), equalTo("test-" + DateFormatter.forPattern("yyyy.MM").format(now) + "-000002"));
         assertThat(response.isDryRun(), equalTo(false));
         assertThat(response.isRolledOver(), equalTo(true));
         assertThat(response.getConditionStatus().size(), equalTo(0));
 
         response = client().admin().indices().prepareRolloverIndex("test_alias").get();
-        assertThat(response.getOldIndex(), equalTo("test-" + DateFormatter.forPattern("YYYY.MM").format(now) + "-000002"));
-        assertThat(response.getNewIndex(), equalTo("test-" + DateFormatter.forPattern("YYYY.MM").format(now) + "-000003"));
+        assertThat(response.getOldIndex(), equalTo("test-" + DateFormatter.forPattern("yyyy.MM").format(now) + "-000002"));
+        assertThat(response.getNewIndex(), equalTo("test-" + DateFormatter.forPattern("yyyy.MM").format(now) + "-000003"));
         assertThat(response.isDryRun(), equalTo(false));
         assertThat(response.isRolledOver(), equalTo(true));
         assertThat(response.getConditionStatus().size(), equalTo(0));
 
         GetSettingsResponse getSettingsResponse = client().admin().indices().prepareGetSettings(response.getOldIndex(),
             response.getNewIndex()).get();
-        assertEquals("<test-{now/M{YYYY.MM}}-000002>", getSettingsResponse.getSetting(response.getOldIndex(),
+        assertEquals("<test-{now/M{yyyy.MM}}-000002>", getSettingsResponse.getSetting(response.getOldIndex(),
             IndexMetaData.SETTING_INDEX_PROVIDED_NAME));
-        assertEquals("<test-{now/M{YYYY.MM}}-000003>", getSettingsResponse.getSetting(response.getNewIndex(),
+        assertEquals("<test-{now/M{yyyy.MM}}-000003>", getSettingsResponse.getSetting(response.getNewIndex(),
             IndexMetaData.SETTING_INDEX_PROVIDED_NAME));
 
         response = client().admin().indices().prepareRolloverIndex("test_alias").setNewIndexName("<test-{now/d}-000004>").get();
-        assertThat(response.getOldIndex(), equalTo("test-" + DateFormatter.forPattern("YYYY.MM").format(now) + "-000003"));
-        assertThat(response.getNewIndex(), equalTo("test-" + DateFormatter.forPattern("YYYY.MM.dd").format(now) + "-000004"));
+        assertThat(response.getOldIndex(), equalTo("test-" + DateFormatter.forPattern("yyyy.MM").format(now) + "-000003"));
+        assertThat(response.getNewIndex(), equalTo("test-" + DateFormatter.forPattern("yyyy.MM.dd").format(now) + "-000004"));
         assertThat(response.isDryRun(), equalTo(false));
         assertThat(response.isRolledOver(), equalTo(true));
         assertThat(response.getConditionStatus().size(), equalTo(0));

--- a/server/src/test/java/org/elasticsearch/cluster/metadata/DateMathExpressionResolverTests.java
+++ b/server/src/test/java/org/elasticsearch/cluster/metadata/DateMathExpressionResolverTests.java
@@ -91,7 +91,6 @@ public class DateMathExpressionResolverTests extends ESTestCase {
                 + DateTimeFormat.forPattern("YYYY.MM.dd").print(new DateTime(context.getStartTime(), UTC).withDayOfMonth(1))));
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/37037")
     public void testExpression_CustomFormat() throws Exception {
         List<String> results = expressionResolver.resolve(context, Arrays.asList("<.marvel-{now/d{yyyy.MM.dd}}>"));
         assertThat(results.size(), equalTo(1));
@@ -106,7 +105,6 @@ public class DateMathExpressionResolverTests extends ESTestCase {
             equalTo(".mar{v}el-" + DateTimeFormat.forPattern("yyyy.MM.dd").print(new DateTime(context.getStartTime(), UTC))));
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/37037")
     public void testExpression_EscapeDateFormat() throws Exception {
         List<String> result = expressionResolver.resolve(context, Arrays.asList("<.marvel-{now/d{'\\{year\\}'yyyy}}>"));
         assertThat(result.size(), equalTo(1));
@@ -127,7 +125,6 @@ public class DateMathExpressionResolverTests extends ESTestCase {
             DateTimeFormat.forPattern("YYYY.MM").print(new DateTime(context.getStartTime(), UTC).withDayOfMonth(1))));
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/37037")
     public void testExpression_CustomTimeZoneInIndexName() throws Exception {
         DateTimeZone timeZone;
         int hoursOffset;

